app/assets/javascripts/course/user_achievements.js

```diff
@@ -0,0 +1,12 @@
+(function($) {
+  'use strict';
+  $('#toggle-on-button' ).click(function(e) {
```

Comment:
Since this is script, the `#toggle-on-button` also needs to be shown by script, like the assessment discussion toolbar.

Also, you'd need to qualify the toggle-on-button to your page, using the same controller.action CSS classes.